manager/src/test/java/com/redhat/service/smartevents/manager/api/user/validators/processors/GatewayConstraintValidatorTest.java

```diff
@@ -0,0 +1,281 @@
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Supplier;
+import java.util.stream.Stream;
+
+import javax.inject.Inject;
+
+import org.hibernate.validator.constraintvalidation.HibernateConstraintValidatorContext;
+import org.hibernate.validator.constraintvalidation.HibernateConstraintViolationBuilder;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.ArgumentCaptor;
+import org.mockito.ArgumentMatchers;
+import org.mockito.verification.VerificationMode;
+
+import com.redhat.service.smartevents.infra.exceptions.definitions.user.GatewayProviderException;
+import com.redhat.service.smartevents.infra.models.gateways.Action;
+import com.redhat.service.smartevents.infra.models.gateways.Gateway;
+import com.redhat.service.smartevents.infra.models.gateways.Source;
+import com.redhat.service.smartevents.infra.validations.ValidationResult;
+import com.redhat.service.smartevents.manager.api.models.requests.ProcessorRequest;
+import com.redhat.service.smartevents.processor.GatewayConfigurator;
+import com.redhat.service.smartevents.processor.GatewayValidator;
+
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit.mockito.InjectMock;
+
+import static com.redhat.service.smartevents.manager.api.user.validators.processors.GatewayConstraintValidator.GATEWAY_CLASS_PARAM;
+import static com.redhat.service.smartevents.manager.api.user.validators.processors.GatewayConstraintValidator.GATEWAY_PARAMETERS_MISSING_ERROR;
+import static com.redhat.service.smartevents.manager.api.user.validators.processors.GatewayConstraintValidator.GATEWAY_PARAMETERS_NOT_VALID_ERROR;
+import static com.redhat.service.smartevents.manager.api.user.validators.processors.GatewayConstraintValidator.GATEWAY_TYPE_MISSING_ERROR;
+import static com.redhat.service.smartevents.manager.api.user.validators.processors.GatewayConstraintValidator.GATEWAY_TYPE_NOT_RECOGNISED_ERROR;
+import static com.redhat.service.smartevents.manager.api.user.validators.processors.GatewayConstraintValidator.MISSING_GATEWAY_ERROR;
+import static com.redhat.service.smartevents.manager.api.user.validators.processors.GatewayConstraintValidator.MULTIPLE_GATEWAY_ERROR;
+import static com.redhat.service.smartevents.manager.api.user.validators.processors.GatewayConstraintValidator.TYPE_PARAM;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.AdditionalMatchers.not;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.reset;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@QuarkusTest
```

Comment:
Does this really need to be a Quarkus test!?!?

You could simply annotate with `@ExtendWith(MockitoExtension.class)`, `@Mock GatewayConfigurator gatewayConfiguratorMock` and manually construct an instance of `GatewayConstraintValidator`? I guess `@QuarkusTest` eliminates the need for some manual construction; but it'd be a lot quicker to execute without it.

Reply:
Thanks for the trick @manstis . I changed it and it's definitely faster (3 times faster on my machine).